### ISSUE TITLE
Returning `&CxxVector<SharedString>` from C++ FFI gives "not FFI-safe" error

### ISSUE THREAD

**Issue by NickNick** [dtolnay/cxx#857]
Similar to https://github.com/dtolnay/cxx/issues/249, but instead of using the SharedString directly, I get this error when the struct containing the string is returned via a CxxVector:

```
#[cxx::bridge]
pub mod ffi {
  struct SharedString {
    x: String,
  }
  unsafe extern "C++" {
    type View;
    fn get_strings(view: &View) -> &CxxVector<SharedString>;
  }
}
```

**Comment by dtolnay** [comment]
Good catch -- I've published a fix in 1.0.49.

**Comment by NickNick** [comment]
That was very quick, thanks!
